### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by ebuchman**

### CODE REVIEW

File: hd/address.go [hunk 1]
```diff
@@ -282,9 +282,20 @@ func CalcSha512(buf []byte) []byte {
 }
 
 func ReverseBytes(buf []byte) []byte {
-	res := []byte{}
-	for i := len(buf) - 1; i >= 0; i-- {
-		res = append(res, buf[i])
+	var res []byte
```
Comment (ebuchman) [inline hd/address.go:285]: Maybe this belongs in tmlibs/common/bytes.go?

File: keys/cryptostore/holder.go [hunk 2]
```diff
@@ -43,12 +43,21 @@ func (s Manager) Create(name, passphrase, algo string) (keys.Info, string, error
...
+	key := gen.Generate(secret)
 	err = s.es.Put(name, passphrase, key)
 	if err != nil {
 		return keys.Info{}, "", err
 	}
-	seed, err := s.codec.BytesToWords(key.Bytes())
+
+	// we append the type byte to the serialized secret to help with recovery
+	// ie [secret] = [secret] + [type]
+	typ := key.Bytes()[0]
```
Comment (ebuchman) [inline keys/cryptostore/holder.go:57]: seems kind of hacky. is there not a better way to get the type byte?

**Review by ebuchman — COMMENTED** [review]
Some nice cleanup

File: keys/cryptostore/holder.go (hunk 2 above)
Comment (ethanfrey) [reply keys/cryptostore/holder.go:57]: there isn't, i was thinking to add this to go-wire, but it is not public afaik

File: hd/address.go (hunk 1 above)
Comment (ethanfrey) [reply hd/address.go:285]: Yeah, a nice reverse function... sounds good

**Pull request merged**